src/main/java/uk/ac/diamond/ispyb/dao/IspybToolsDAO.java

```diff
@@ -0,0 +1,135 @@
+        Map<String, Object> map = new HashMap<>();
+        map.put("dataCollectionId", dataCollectionId);
+        map.put("imageNumber", imageNumber);
+        return templateWrapper.callIspybForBean("retrieve_quality_indicators", ImageQualityIndicators.class, map);
+    }
+
+    @Override
+    public  List<Screening> retrieveScreeningsForSample(Long sampleId) throws SQLException {
+        return templateWrapper.callIspybForListBeans("retrieve_screenings_for_sample", Screening.class, map("id", sampleId));
+    }
+
+    @Override
+    public  List<DataCollectionMain> retrieveDataCollectionsForSample(Long sampleId) throws SQLException {
+        return templateWrapper.callIspybForListBeans("retrieve_dcs_for_sample", DataCollectionMain.class, map("id", sampleId));
+    }
+
+    @Override
+    public  Optional<DataCollection> retrieveDataCollection(Long dataCollectionId, String authLogin) throws SQLException {
+        Map<String, Object> params = new HashMap<>();
+        params.put("id", dataCollectionId);
+        params.put("authLogin", authLogin);
+        return templateWrapper.callIspybForBean("retrieve_dc", DataCollection.class, params);
+    }
+
+    @Override
+    public  Optional<MxSample> retrieveSample(Long sampleId, boolean useContainerSession, String authLogin) throws SQLException {
+        Map<String, Object> _map = new HashMap<>();
+        _map.put("id", sampleId);
+        _map.put("useContainerSession", useContainerSession);
+        _map.put("authLogin", authLogin);
+        return templateWrapper.callIspybForBean("retrieve_sample", MxSample.class, _map);
+    }
+
+    @Override
+    public  Optional<Container> retrieveContainer(Long containerId, boolean useContainerSession, String authLogin) throws SQLException {
+        Map<String, Object> _map = new HashMap<>();
+        _map.put("id", containerId);
+        _map.put("useContainerSession", useContainerSession);
+        _map.put("authLogin", authLogin);
+        return templateWrapper.callIspybForBean("retrieve_container", Container.class, _map);
+    }
+
+    @Override
+    public Long insertAperture(Aperture aperture) throws SQLException {
+        return beanTemplateWrapper.callIspybForKey("insert_aperture", Long.class, aperture, "p_id").get();
+    }
+
+    @Override
+    public Long insertDataCollectionGroup(DataCollectionGroup dataCollectionGroup) throws SQLException {
+        return beanTemplateWrapper.callIspybForKey("upsert_dc_group_v3", Long.class, dataCollectionGroup, "p_id").get();
+    }
+
+    @Override
+    public Long insertDataCollection(DataCollectionMain dataCollectionMain) throws SQLException {
+        return beanTemplateWrapper.callIspybForKey("upsert_dc_main_v3", Long.class, dataCollectionMain, "p_id").get();
+    }
+
+    @Override
+    public Long updateSample(Sample sample) throws SQLException {
+        return beanTemplateWrapper.callIspybForKey("upsert_sample", Long.class, sample, "p_id").get();
```

Comment:
The existing upsert_sample sproc is using a slightly different set of columns and different aliases for them, so I'm not sure this call would work too well.

Reply:
Yeah, I'll probably have to write a new sproc for that. I also need to change it to use MxSample most likely.

Reply:
OK, maybe just make it a call for a yet-to-be-created "upsert_mx_sample" for now, and then we'll sort out creating that + tests later.